lib/Module/Metadata.pm

```diff
@@ -10,7 +10,7 @@ package Module::Metadata;
 # perl modules (assuming this may be expanded in the distant
 # parrot future to look at other types of modules).
 
-sub __clean_eval { eval $_[0] }
+sub __clean_eval { no strict; no warnings; eval $_[0] }
```

Comment:
I'm mostly ok with this, but just would like to note that "no warnings" does not currently bring things to the "default" state, it in fact disables default warnings. However, I don't believe there is any reason for this module to care about default warnings here either.

Reply:
Is there a specific warning category that is being targeted here? If so, can't we just disable that one?

In other words, what forms of syntax are being tripped up by having warnings enabled?

Reply:
I don't think this is meant to guard against anything specific, but against the possibility of warnings being enabled in an outer scope.

Reply:
I was simply trying to explicitly do what I guessed the intentions were by declaring this sub prior to use warnings/strict.

Reply:
And as far as I know there is no way to declare that a scope should use what is currently the default set of warnings, which is an unfortunate limitation of the warnings.pm api.

Reply:
For context: https://perldoc.pl/warnings#Default-Warnings-and-Optional-Warnings

Reply:
IMO the current change make the original intent clear.

The function `__clean_eval` as it stands (before this fix) is eval without warnings and strict hint set.
With this patch, it stays the same but would avoid someone to figure out what's happening and make the intent clear.

Yes this is hiding some issues that would have been exposed with strict and warnings, but I think this is a different case, no?

That code runs without warnings and strict being set.
```perl
sub alien { $x = "2:" + 3; return $x }

use strict;
use warnings;

print alien() . "\n";
```

Reply:
Correct. I am pointing out that it is not in fact the same as without warnings and strict set. "no warnings" also disables default warnings.

Reply:
> Correct. I am pointing out that it is not in fact the same as without warnings and strict set. "no warnings" also disables default warnings.

What do you suggest instead?

Reply:
I would note that this sub is only used in 2 places and the change that created it seemed to want to turn off all strict/warnings regardless. I didn't write it so I can only speculate on the intention.

Reply:
If I had a better suggestion I would not have approved it :wink: